### PR TITLE
Refactor: use SDK `amount.Amount` type to store token amounts

### LINKED ISSUE #2853: Refactor balances, store them as `amount.Amount` not string
bigint good. bigint no have dumb bug

### PR DESCRIPTION
**Pull request by artursapek**
Fixes #2853

## Changes

- Change `Balance` type, and `amount` values in Redux, to store an `amount.Amount`, not a `number`
- Update validation code accordingly
- Remove formatting hacks (removing commas, etc)
- Remove `formatAmount` helper function (not required anymore)
- Remove dead code in SDKv2 adapter

Overall cleans things up and net removal of code

https://www.loom.com/share/3b0d63df72c04b7182a41e0ca173e9cc

### CODE REVIEW

File: wormhole-connect/src/hooks/useAmountValidation.ts [hunk 1]
```diff
@@ -49,37 +49,30 @@ export const useAmountValidation = (props: Props): HookReturn => {
     [props.quotesMap],
   );
 
-  const allRoutesFailed = useMemo(
-    () =>
-      props.routes.every((route) => props.quotesMap[route]?.success === false),
-    [props.routes, props.quotesMap],
-  );
+  const allRoutesFailed = useMemo(() => {
+    if (Object.keys(props.quotesMap).length === 0) {
```
Comment (artursapek) [inline wormhole-connect/src/hooks/useAmountValidation.ts:53]: this fixes a bug unrelated to `amount.Amount` refactor: if you run `every` on an empty array, it always returns `true`.

File: wormhole-connect/src/hooks/useFetchSupportedRoutes.ts [hunk 2]
```diff
@@ -39,7 +39,6 @@ const useFetchSupportedRoutes = (): HookReturn => {
           supported = await route.isRouteSupported(
             token,
             destToken,
-            amount,
```
Comment (artursapek) [inline wormhole-connect/src/hooks/useFetchSupportedRoutes.ts:43]: this argument was never used by `isRouteSupported`

File: wormhole-connect/src/routes/types.ts [hunk 3]
```diff
@@ -1,37 +0,0 @@
-import { TransferInfo } from '../utils/sdkv2';
```
Comment (kev1n-peters) [inline wormhole-connect/src/routes/types.ts]: Should we delete this file since it's now empty?

Comment (artursapek) [reply wormhole-connect/src/routes/types.ts]: thought I had. thanks

File: wormhole-connect/src/utils/index.ts [hunk 4]
```diff
@@ -135,24 +139,18 @@ export function getGasToken(chain: Chain): TokenConfig {
...
-  chain: Chain,
-  tokenId: TokenId | 'native' = 'native',
-): number {
+export function getTokenDecimals(chain: Chain, token: TokenConfig): number {
   const chainConfig = config.chains[chain];
   if (!chainConfig) throw new Error(`chain config for ${chain} not found`);
 
-  if (tokenId === 'native') {
+  /*
+  if (token?.tokenId === 'native') {
```
Comment (emreboga) [inline wormhole-connect/src/utils/index.ts:147]: Forgot to remove this part?

**Review by kev1n-peters — APPROVED** [review]

**Pull request merged**